cms/djangoapps/contentstore/management/commands/set_advanced_fields_all_courses.py

```diff
@@ -0,0 +1,248 @@
+            }, \
+            "invitation_only": { \
+                "value": "true" \
+            } \
+        }'
+    user_help = '--user <email> required, needs to be a staff member'
+    course_help = '--course <id> required, e.g. course-v1:ORG+COURSENUMBER+COURSERUN'
+    force_update_catalog_visibility_about_help = '--force-update-catalog-visibility-about, To fix `There was an error loading this course` error when loading MFE learning this forces `catalog_visibility` from `none` to `about`.'  # pylint: disable=line-too-long
+    limit_bulk_update_operation_help = '--limit-bulk-update-operation {0-N}'
+
+    _force_update_catalog_visibility_about = False
+    _updated_catalog_visilibity_about = []
+
+    def add_arguments(self, parser):
+        parser.add_argument('--fields_json',
+                            dest='fields_json',
+                            default=False,
+                            required=True,
+                            help=self.fields_json_help)
+        parser.add_argument('--user',
+                            dest='user_email',
+                            default=False,
+                            required=True,
+                            help=self.user_help)
+        parser.add_argument('--course',
+                            dest='course_id',
+                            default=False,
+                            required=False,
+                            help=self.course_help)
+        parser.add_argument('--force-update-catalog-visibility-about',
+                            dest='force_update_catalog_visibility_about',
+                            action='store_true',
+                            help=self.force_update_catalog_visibility_about_help)
+
+        # WARNING
+        # Limiting the all courses bulk advanced setting update here to avoid consuming all
+        # server memory. Defaulting bulk operations to 150 courses but you can use this argument
+        # to specify any value different than the default.
+        parser.add_argument('--limit-bulk-update-operation',
+                            dest='limit_bulk_update_operation',
+                            default=150,
+                            required=False,
+                            help=self.limit_bulk_update_operation_help)
+
+    def handle(self, *args, **options):
+        """
+        Execute the command
+        """
+        self._force_update_catalog_visibility_about = options.get(
+            'force_update_catalog_visibility_about', False
+        )
+        try:
+            update_fields = json.loads(options["fields_json"])
+        except ValueError as err:
+            raise CommandError("Invalid JSON object") from err
+
+        # Update course(s) to push out `fields_json` of advanced setting changes.
+        self._set_courses_advanced_fields(
+            update_fields,
+            options['user_email'],
+            options['course_id'],
+            int(options['limit_bulk_update_operation'])
```

Comment:
Argument `limit_bulk_update_operation` should have `type=int` such that we don't have to type-case to int here.